zebedee-cms/src/test/java/com/github/onsdigital/zebedee/keyring/CollectionKeyringImplTest.java

```diff
@@ -158,6 +169,49 @@ public void testPopulateFromUser_success() throws Exception {
         keyring.populateFromUser(user);
 
         // Then the central keyring is updated with each entry in the user keyring.
-        verify(cache, times(1)).add(TEST_COLLECTION_ID, secretKey);
+        verify(keyringCache, times(1)).add(TEST_COLLECTION_ID, secretKey);
+    }
+
+    @Test
+    public void testGetInstance_notInitialised() throws Exception {
+        // Given CollectionKeyring has not been initalised
+        resetInstanceToNull();
+
+        // When GetInstance is called
+        // Then an exception is thrown
+        KeyringException ex = assertThrows(KeyringException.class, () -> CollectionKeyringImpl.getInstance());
+        assertThat(ex.getMessage(), equalTo(NOT_INITALISED_ERR));
+    }
+
+    @Test
+    public void testGetInstance_success() throws KeyringException {
+        // Given CollectionKeyring has been initalised
+
+        // When GetInstance is called
+        CollectionKeyring collectionKeyring = CollectionKeyringImpl.getInstance();
+
+        // Then a non null instance is returned
+        assertThat(collectionKeyring, is(notNullValue()));
+    }
+
+    @Test
+    public void testInit_keyringCacheNull() throws Exception {
+        // Given keyringCache is null
+        KeyringCache keyringCache = null;
+
+        resetInstanceToNull();
+
+        // When init is called
+        KeyringException ex = assertThrows(KeyringException.class, () -> CollectionKeyringImpl.init(keyringCache));
+
+        // Then an exception is thrown
+        assertThat(ex.getMessage(), equalTo("keyringCache required but was null"));
```

Comment:
Should this use the KEYRING_CACHE_NULL_ERR const from CollectionKeyringImpl?

Reply:
Yes, this is a fair point. It should.